src/actions/actions.ts

```diff
@@ -3673,6 +3673,11 @@ abstract class MoveByScreenLine extends BaseMovement {
   value: number = 1;
 
   public async execAction(position: Position, vimState: VimState): Promise<Position | IMovement> {
+    if (vimState.currentMode === ModeName.VisualLine) {
```

Comment:
So the main issue is that we don't handle visualLine selections, due to the fact that our cursor must be at either start or stop (which as far as I can tell, works for every case but this one?).

However, in vim in visual line mode, doing gj on a wrapped line causes you to maintain the same highlighted area, however, your cursor is moved up one wrapped line.

However, we always snap the cursors at the beginning and the end of the selection to the end of the line (to maintain visual line mode). Thus, if we do a "gj" on a wrapped line, we are snapped back to our initial position, and we can't move down.

I don't think there's a good solution for this, so the workaround I came up with was to move by a "line" instead of a "wrappedLine" if you're doing visualLine movement.

This behavior doesn't exactly match vim, but I really doubt (I hope) that anybody relied on that feature of vim.